### PR TITLE
WIP: Add tof

### PR DESCRIPTION
**Pull request by gaede**
BEGINRELEASENOTES
- add new package TimeOfFlight
     - use TOFEstimators processor to compute TOF parameters
     - will be added as PID object to the ReconstructedParticles

ENDRELEASENOTES

### CODE REVIEW

File: TimeOfFlight/src/TOFEstimators.cc [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+
+      
+      // =======  use only Ecal hits  (requires the CalorimeterHitType to be set in the digitizer )
+      //          with time information ( > 1 ps) and layer <= max layer
+
+      int maxLayerNum = _maxLayerNum ;
+      std::function<bool(CalorimeterHit*)> selectHits =  [maxLayerNum](CalorimeterHit* h){
+
+	return ( isEcal( h )            &&
```
Comment (rete) [inline TimeOfFlight/src/TOFEstimators.cc]: Why do you use only ecal hits ? What about neutral hadrons ?

File: TimeOfFlight/src/TOFEstimators.cc [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+    colPFO = nullptr ;
+  }
+    
+  if( colPFO != nullptr ){
+    
+
+
+    PIDHandler pidh( colPFO );
+    int algoID = pidh.addAlgorithm( name()  , _TOFNames);
```
Comment (rete) [inline TimeOfFlight/src/TOFEstimators.cc]: Maybe we should add a processor parameter for this. If it is not provided, we can jump to a fallback with the processor name. What do you think ?

File: TimeOfFlight/src/TOFEstimators.cc (hunk 1 above)
Comment (gaede) [reply TimeOfFlight/src/TOFEstimators.cc]: At this point it is a feature that we restrict ourselves to the Ecal only. Combining hits from Ecal and Hcal with different time resolutions would probably require a more involved algorithm.
We can leave this for a future version.

File: TimeOfFlight/src/TOFEstimators.cc (hunk 2 above)
Comment (gaede) [reply TimeOfFlight/src/TOFEstimators.cc]: Actually, I think this is more elegant as the Marlin framework requires that different processors of the same type have different names. If you had an additional parameter you'd have to change both consistently in the steering file. What would be the advantage ?

**Pull request merged**